samples/chaincode/cc-tools-demo/README.md

```diff
@@ -0,0 +1,274 @@
+
+This tutorial is based on the [FPC with CC-Tools integration project]() and all our design choices are explained here in the [design document]()
+Here are the steps to accomplish this:
+
+* Clone and copy the cc-tools-demo chaincode
+* Edit the chaincode to became an FPC chaincode instead of normal fabric
+* Build your FPC Go Chaincode
+* Launch a Fabric network
+* Install and instantiate your chaincode
+* Invoke transactions by using the FPC simple-cli
+
+## Prerequisites
+
+This tutorial presumes that you have installed FPC on your `$GOPATH` as described in the FPC [README.md](../../../README.md#clone-fabric-private-chaincode) and `$FPC_PATH` is set accordingly.
+
+We also need a working FPC development environment. As described in the "Setup your Development Environment" Section of the FPC [README.md](../../../README.md#setup-your-development-environment), you can use our docker-based dev environment (Option 1) or setup your local development environment (Option 2).
+We recommend using the docker-based development environment and continue this tutorial within the dev container terminal.
+
+Moreover, within your FPC development you have already installed the FPC Go Chaincode Support components.
+See the installation steps in [ecc_go/README.md](../../../ecc_go/README.md#installation).
+
+We also assume that you are familiar with Fabric chaincode development in go.
+Most of the steps in this tutorial follow the normal Fabric chaincode development process, however, there are a few differences that we will highlight here.
+
+## Clone and copy the cc-tools-demo chaincode
+
+Clone the [cc-tools-demo]() repository and copy the [chaincode]() folder. Then paste it in the root directory for cc-tools-demo here.
+
+```bash
+cd ~
+git clone https://github.com/hyperledger-labs/cc-tools-demo.git
+cp -a ~/cc-tools-demo/chaincode/. $FPC_PATH/samples/chaincode/cc-tools-demo/
+cd $FPC_PATH/samples/chaincode/cc-tools-demo
+```
+
+The chaincode code structure is different than normal chaincode as it's using the cc-tools framework
+
+**Note**: remove all test files (`*_test.go`) you find like `txdefs_createNewLibrary_test.go` as they are not part of the chaincode.
+
+## Edit the chaincode to became an FPC chaincode instead of normal fabric
+
+Go to `$FPC_PATH/samples/chaincode/cc-tools-demo/main.go` and create the project structure.
+
+```bash
+cd $FPC_PATH/samples/chaincode/cc-tools-demo
+```
+
+The code presented here is not different from traditional Go Chaincode developed with CC-Tools framework. All the FPC specific protection mechanisms are handled by the FPC framework transparently.
+
+The `main.go` contains the starting point of the chaincode.
+
+To use FPC, we need to add some logic to instantiate our private chaincode and start it.
+To do so, we use `fpc.NewPrivateChaincode(&chaincode.AssetExample{})` and since we're already in the FPC repo, remove the `go.mod` and `go.sum` files so we use the chaincode as part of the FPC package.
```

Comment:
I think this should be something like ... `shim.Start(fpc.NewPrivateChaincode(new(CCDemo)))`

Reply:
Done